src/ui/ui_i18n/README.md

```diff
@@ -0,0 +1,169 @@
+```
+
+Creating instances of `IntlMessageFormat` is an expensive operation.
+[Intl-format-cache](https://github.com/yahoo/intl-format-cache)
+library is simply to make it easier to create a cache of format
+instances of a particular type to aid in their reuse. Under the
+hood, this package creates a cache key based on the arguments passed
+to the memoized constructor.
+
+```
+import memoizeIntlConstructor from 'intl-format-cache';
+
+const getMessageFormat = memoizeIntlConstructor(IntlMessageFormat);
+```
+
+`Intl-messageformat` package assumes that the
+[Intl](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Intl)
+global object exists in the runtime. `Intl` is present in all modern
+browsers and Node.js > 0.10. In order to load `Intl MessageFormat`
+in Node.js we should simply `require()` this package (in Node.js,
+the data for all 200+ languages is loaded along with the library):
+```
+const IntlMessageFormat = require('intl-messageformat');
+```
+
+After that we are able to use this library in the same way as for browsers.
+
+## Angular wrapper
+
+Angular wrapper will have at least 3 entities: translation `service`,
+`directive` and `filter`. Both, the directive and the filter will use
+the translation service with `intl-messageformat` library under the hood.
+
+The translation `service` will provide the following methods (names can be changed):
+- `addMessages(messages: object, [locale: string])` - provides a way to register
+translations with the library
+- `getMessages()` - returns messages for the current language
+- `locale(locale: string)` - tells the library which language to use by given
+language key
+- `locale()` - returns the current locale 
+- `defaultLocale(locale: string)` - tells the library which language to fallback
+when missing translations
+- `defaultLocale()` - returns the default locale
+- `defineFormats(formats: object)` - supply a set of options to the underlying formatter
+- `translate({id: string, values: object, defaultMessage: string})` – translate message by id
+
+The translation `filter` will look like this:
+```
+{{'translationId' | i18n[:{ values: object, defaultMessage: string }]}}
+```
+
+Where:
+- `translationId` - translation id to be translated
+- `values` - values to pass into translation
+- `defaultMessage` - will be used unless translation was successful
+
+The translation `directive` will look like this:
+```
+<ANY
+  i18n="{string}"
+  [i18n-values="{object}"]
+  [i18n-default-message="{string}"]
+></ANY>
+```
+
+Where:
+- `i18n` - translation id to be translated
+- `i18n-values` - values to pass into translation
+- `i18n-default-message` - will be used unless translation was successful
+
+## Localization files
+
+Localization files will have [JSON5](https://github.com/json5/json5) format,
+so single and multi-line comments are allowed. It can help to understand
+which section of application localization key is used for. Also `namespaces`
```

Comment:
This sentence confuses me a little. Do you mean 
> It can help to understand which section of the application the localization key is used for

or 

> It can help to understand which section of application localization the key is used for

or..

There are a number of places where a "the" could help, but I'm not sure what you mean, so not sure where it goes.

Reply:
__optional nit:__ Also it'd probably be better if you can enumerate (as list items maybe?) main benefits of `JSON5` so that it's easy to understand why we decided to use it.

Reply:
:+1: for the "whys"